test/integration/options/test_configure_options.py

```diff
@@ -123,3 +123,32 @@ class Pkg(ConanFile):
         client.save({"conanfile.py": conanfile})
         client.run(f"create . --name=pkg --version=0.1")
         self.assertIn("Package 'da39a3ee5e6b4b0d3255bfef95601890afd80709' created", client.out)
+
+
+def test_possible_values_based_on_version():
+    tc = TestClient(light=True)
+    tc.save({"conanfile.py": textwrap.dedent("""
+    from conan import ConanFile
+    from conan.tools.scm import Version
+
+    class Pkg(ConanFile):
+        name = "foo"
+        options = {"my_option": ["foo", "v1", "v2"]}
+        default_options = {"my_option": "foo"}
+        def config_options(self):
+            if Version(self.version) < "2.0":
```

Comment:
What would be the advantage of this instead of raising an InvalidConfiguration in ``validate()`` for the non-possible values for version < 2.0?